### PR TITLE
Upstream community kind cluster setup

### PR DESCRIPTION
**Pull request by mvalarh**
Signed-off-by: Martin Vala <mavala@redhat.com>

### CODE REVIEW

**Comment by wcheang** [comment]
I haven't finished looking through everything yet, but some thoughts so far:

- There are a bunch of lines that are commented out that look like they might be needed. Please go through and make sure any lines that are needed are uncommented. If they're not needed please remove them.
- Can the `base` role be renamed to something more specific (`upstream-community-base` perhaps?), just to be clear that it's not a base for all ansible roles.
- Do all the builtins actually have to start with `ansible.builtin.`? I haven't really seen required before.

**Comment by mvalarh** [comment]
> 
Can the base role be renamed to something more specific (upstream-community-base perhaps?), just to be clear that it's not a base for all ansible roles.

Done

**Comment by mvalarh** [comment]
> There are a bunch of lines that are commented out that look like they might be needed. Please go through and make sure any lines that are needed are uncommented. If they're not needed please remove them.

I will start to remove them tomorrow

File: ansible/playbooks/upstream-community.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  tasks:
+    - name: "Base"
+      import_role:
+        name: base
+      tags:
+        - always
+
+    - name: "Install prerequisites"
+      include_role:
+        name: install-prerequisites
```
[the remark this answers is not in the document]
Comment (mvalarh) [reply ansible/playbooks/upstream-community.yaml]: Done

**Comment by amisstea** [comment]
> @wcheang @amisstea Do you want me to provide variable `run_upstream_community` default to false and we will set it to true?

It may not be necessary since the kind cluster setup shouldn't block the workflows of the ISV pipelines, right? If you think it will slow down the overall CI/CD then please do add it.

**Comment by mvalarh** [comment]
> > @wcheang @amisstea Do you want me to provide variable `run_upstream_community` default to false and we will set it to true?
> 
> It may not be necessary since the kind cluster setup shouldn't block the workflows of the ISV pipelines, right? If you think it will slow down the overall CI/CD then please do add it.

I think it is not needed now. Let's see if we will need it in future. If you merge this i will try to parametrize images in tasks, so they are not hard-coded there. Thanks for your time

**Comment by J0zi** [comment]
@amisstea what is the process to get it merge? Anything blocking us?

**Comment by amisstea** [comment]
> @amisstea what is the process to get it merge? Is anything blocking us?

@J0zi I'll merge this as soon as @wcheang approves.

**Review by wcheang — APPROVED** [review]

File: ansible/roles/base/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    path: "{{ op_bin_dir }}"
+    mode: "755"
+  with_items:
+    - directory
+
+- name: "Load config '{{ oc_config_file }}'"
+  ansible.builtin.include_role:
+    name: include-vars
+  vars:
+    var_name: uc_config
```
[the remark this answers is not in the document]
Comment (mvalarh) [reply ansible/roles/base/tasks/main.yml]: uc (upstream community)

[the remark this answers is not in the document]
Comment (mvalarh) [reply ansible/roles/base/tasks/main.yml]: Maybe i can use local variable name for roles as starting initials if we want. Example, `upstream-community-base` role would have variables starting with `ucb_`

[the remark this answers is not in the document]
Comment (wcheang) [reply ansible/roles/base/tasks/main.yml]: Ah, `uc` standing for `upstream community` makes sense. I was just wondering and wanted to make sure it wasn't a typo. It's fine as-is.

**Comment by mvalarh** [comment]
@amisstea i think it is ready to go.

**Pull request merged**